### PR TITLE
Fix `TouchStickType::Dynamic` stuck in upper left corner when not dragging

### PR DESCRIPTION
**Pull request by Hellzbellz123**
base_position should probably be called last_drag_positon

### CODE REVIEW

**Comment by Hellzbellz123** [comment]
I have some documentation on the if statement for get_base_pos() when the touchstick is dynamic
I did not push it because im not sure if it's necessary.
if it feels like magic i can re add it

the first if is if not dragging
the second if statement is for when the mouse leaves the window

**Comment by johanhelsing** [comment]
In this pr, you merged the wrong branch, I think (your other pr branch, and not my `main` branch).

There are two commits that do exactly the same change: 6858f73 and bf3948c .

Could you do a rebase and keep only one of them, then force-push?

Here's how you do it:

```shell
git rebase -i johanhelsing/main
```

Replace `johanhelsing` with whatever you called my remote. From the merge commit above, I'm guessing it's not origin.

Then your text editor opens, and a file with the two commits open. Delete one of the lines, save and exit. Or use equivalent gui tools to do it.

If you do `git log` now, you should see a single commit on top of main. Now do a force push:

```
git push -f
```

**Comment by Hellzbellz123** [comment]
i hope i did that correctly. sorry for the trouble.
git has definitely been a learning experience

File: src/ui.rs [hunk 1]
```diff
@@ -159,6 +159,12 @@ fn get_base_pos<S: StickIdType>(
...
             }
         }
-        TouchStickType::Dynamic => stick.base_position.extend(0.),
+        TouchStickType::Dynamic => {
+            if stick.drag_id.is_none() || stick.base_position == Vec2::ZERO {
+                global_transform.translation()
+            } else {
+                stick.base_position.extend(0.)
+            }
+        } //stick.base_position.extend(0.),
```
Comment (johanhelsing) [inline src/ui.rs]:
```suggestion
        }
```

**Review by johanhelsing — APPROVED** [review]
Thanks! history looks good now.

**Pull request merged**